File: snippets/rust/src/self_signer.rs
use std::sync::Arc;
use std::path::PathBuf;
use std::fs;

use anyhow::Result;
use bip39::{Language, Mnemonic};
use breez_sdk_liquid::prelude::*;
use log::{info};

// ANCHOR: self-signer  
async fn connect_with_self_signer(signer: Signer) -> Result<Arc<LiquidSdk>> {  
  // Create the default config, providing your Breez API key
  let mut config = LiquidSdk::default_config(LiquidNetwork::Mainnet, Some("<your-Breez-API-key>".to_string()))?;

  // Customize the config object according to your needs
  config.working_dir = "path to an existing directory".into();

  let connect_request = ConnectWithSignerRequest {      
      config,
  };
  let sdk = LiquidSdk::connect_with_signer(connect_request, signer, None).await?;

  Ok(sdk)
}
// ANCHOR_END: self-signer

File: snippets/go/self_signer.go
package example

import (
	"github.com/breez/breez-sdk-liquid-go/breez_sdk_liquid"
)

// ANCHOR: self-signer
func ConnectWithSelfSigner(signer breez_sdk_liquid.Signer) (*breez_sdk_liquid.BindingLiquidSdk, error) {

	// Create the default config, providing your Breez API key
	breezApiKey := "<your-Breez-API-key>"
	config, err := breez_sdk_liquid.DefaultConfig(breez_sdk_liquid.LiquidNetworkMainnet, &breezApiKey)
	if err != nil {
		return nil, err
	}

	// Customize the config object according to your needs
	config.WorkingDir = "path to an existing directory"

	connectRequest := breez_sdk_liquid.ConnectWithSignerRequest{
		Config: config,
	}

	sdk, err := breez_sdk_liquid.ConnectWithSigner(connectRequest, signer, nil)

	return sdk, err
}

// ANCHOR_END: self-signer
